test/TextToBinary.Barrier.cpp

```diff
@@ -86,4 +88,101 @@ TEST_F(OpMemoryBarrier, BadInvalidMemorySemanticsId) {
 // TODO(dneto): OpGroupUMax
 // TODO(dneto): OpGroupSMax
 
+// Matches if the SpirvVector under test has length+opcode in the first word of
+// the first instruction.
+MATCHER_P2(StartsWith, opcode, length, "") {
+  return arg[TextToBinaryTest::kFirstInstruction] ==
+         spvOpcodeMake(length, opcode);
+}
+
+using NamedMemoryBarrierTest = spvtest::TextToBinaryTest;
+
+TEST_F(NamedMemoryBarrierTest, OpcodeRecognizedInV11) {
+  EXPECT_THAT(CompileSuccessfully("OpMemoryNamedBarrier %bar %scope %semantics",
+                                  SPV_ENV_UNIVERSAL_1_1),
+              StartsWith(SpvOpMemoryNamedBarrier, 4));
+}
+
+TEST_F(NamedMemoryBarrierTest, OpcodeUnrecognizedInV10) {
+  EXPECT_THAT(CompileFailure("OpMemoryNamedBarrier %bar %scope %semantics",
+                             SPV_ENV_UNIVERSAL_1_0),
+              Eq("Invalid Opcode name 'OpMemoryNamedBarrier'"));
+}
+
+TEST_F(NamedMemoryBarrierTest, ArgumentCount) {
+  EXPECT_THAT(CompileFailure("OpMemoryNamedBarrier", SPV_ENV_UNIVERSAL_1_1),
+              Eq("Expected operand, found end of stream."));
+  EXPECT_THAT(
+      CompileFailure("OpMemoryNamedBarrier %bar", SPV_ENV_UNIVERSAL_1_1),
+      Eq("Expected operand, found end of stream."));
+  EXPECT_THAT(
+      CompileFailure("OpMemoryNamedBarrier %bar %scope", SPV_ENV_UNIVERSAL_1_1),
+      Eq("Expected operand, found end of stream."));
+  EXPECT_THAT(
+      CompileFailure("OpMemoryNamedBarrier %bar %scope %semantics %extra",
+                     SPV_ENV_UNIVERSAL_1_1),
+      Eq("Expected '=', found end of stream."));
+}
+
+TEST_F(NamedMemoryBarrierTest, ArgumentTypes) {
+  EXPECT_THAT(CompileFailure("OpMemoryNamedBarrier 123 %scope %semantics",
+                             SPV_ENV_UNIVERSAL_1_1),
+              Eq("Expected id to start with %."));
+  EXPECT_THAT(CompileFailure("OpMemoryNamedBarrier %bar %scope \"semantics\"",
+                             SPV_ENV_UNIVERSAL_1_1),
+              Eq("Expected id to start with %."));
+}
+
+using TypeNamedBarrierTest = spvtest::TextToBinaryTest;
+
+TEST_F(TypeNamedBarrierTest, OpcodeRecognizedInV11) {
```

Comment:
Will this `OpcodeRecognizedInV11` test appear for each newly added instruction in 1.1? It would be nice to have some kind of fixture to wrap the `OpcodeRecognizedInV11` & `OpcodeUnRecognizedInV10` and supply a complete list using templating.